example/cajita_tutorial/08_array/array_example.cpp

```diff
@@ -0,0 +1,232 @@
+ * Copyright (c) 2018-2021 by the Cabana authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Cabana library. Cabana is distributed under a   *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <Cajita.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <iostream>
+
+#include <mpi.h>
+
+//---------------------------------------------------------------------------//
+// Array example.
+//---------------------------------------------------------------------------//
+void arrayExample()
+{
+    /*
+      The array and corresponding array layout allocates and stores the physical
+      quantities on the grid - prior to creating an Array, only the size, shape,
+      and indexing of the grid is defined.
+    */
+
+    int comm_rank = -1;
+    MPI_Comm_rank( MPI_COMM_WORLD, &comm_rank );
+
+    if ( comm_rank == 0 )
+    {
+        std::cout << "Cajita Array Example" << std::endl;
+        std::cout << "    (intended to be run with MPI)\n" << std::endl;
+    }
+
+    using exec_space = Kokkos::DefaultHostExecutionSpace;
+    using device_type = exec_space::device_type;
+
+    /*
+      The array layout is an intermediate class between the local grid and array
+      - it does not store the mesh field data. Instead it holds the layout of
+      the field arrays.
+
+      As with the previous examples, we define everything necessary to create
+      the local grid.
+    */
+
+    // Let MPI compute the partitioning for this test.
+    Cajita::DimBlockPartitioner<3> partitioner;
+
+    // Create the global mesh.
+    double cell_size = 0.50;
+    std::array<int, 3> global_num_cell = { 37, 15, 20 };
+    std::array<bool, 3> is_dim_periodic = { true, true, true };
+    std::array<double, 3> global_low_corner = { 1.2, 3.3, -2.8 };
+    std::array<double, 3> global_high_corner = {
+        global_low_corner[0] + cell_size * global_num_cell[0],
+        global_low_corner[1] + cell_size * global_num_cell[1],
+        global_low_corner[2] + cell_size * global_num_cell[2] };
+    auto global_mesh = Cajita::createUniformGlobalMesh(
+        global_low_corner, global_high_corner, global_num_cell );
+
+    // Create the global grid.
+    auto global_grid = Cajita::createGlobalGrid( MPI_COMM_WORLD, global_mesh,
+                                                 is_dim_periodic, partitioner );
+
+    /*
+      An array layout includes the local grid size information together with the
```

Comment:
An array layout includes the local grid size information together with the, the number of values per mesh entity in the field (dimensionality), and the entity type on which the field is defined. An array may only be defined on a single entity type.